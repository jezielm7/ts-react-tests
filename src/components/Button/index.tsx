import React from 'react';

// import { Container } from './styles';

const Button: React.FC = ({ label, onClick }) => {
  return (
    <button onClick={onClick}><b>{label}</b></button>
  );
}

export default Button;