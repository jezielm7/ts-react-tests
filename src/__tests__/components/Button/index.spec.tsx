import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from '../../../components/Button';

describe('Test button successfully', () => {
  it('Should render the button successfully', () => {
    const buttonText = 'Login';
    const { container } = render(<Button label={buttonText} />);

    const button = screen.getByRole('button', {name: buttonText});

    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot(); // Snapshot Test
  })
})