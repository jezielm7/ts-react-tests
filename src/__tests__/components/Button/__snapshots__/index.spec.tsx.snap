// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button successfully Should render the button successfully 1`] = `
<div>
  <button>
    <b>
      Login
    </b>
  </button>
</div>
`;
